modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/integration/TableDdlIntegrationTest.java

```diff
@@ -496,6 +496,35 @@ public void createTableAsSelectFromDistributedTable() {
         assertEquals(100L, sql("select count(*) from my_table2").get(0).get(0));
     }
 
+    /**
+     * Creates table with nullabale/not nullable columns and checks nullability.
+     */
+    @Test
+    public void createTableWithNullability() {
+        List<String> dataTypes = F.asList("INT", "FLOAT", "DOUBLE", "DECIMAL", "DECIMAL(10,2)", "VARCHAR",
+            "VARCHAR(10)", "CHAR", "CHAR(10)", "DATE", "TIME", "TIMESTAMP", "BINARY", "BINARY(10)", "VARBINARY",
+            "VARBINARY(10)", "UUID", "OTHER");
+
+        for (String dataType : dataTypes) {
+            try {
+                sql("CREATE TABLE my_table (id INT PRIMARY KEY, val " + dataType + ')');
+                sql("INSERT INTO my_table(id, val) VALUES (0, NULL)");
+            }
+            finally {
+                sql("DROP TABLE my_table");
+            }
+
+            try {
+                sql("CREATE TABLE my_table (id INT PRIMARY KEY, val " + dataType + " NOT NULL)");
+                assertThrows("INSERT INTO my_table(id, val) VALUES (0, NULL)", IgniteSQLException.class,
```

Comment:
Does this duplicate alterTableAddNotNullColumn() or alterTableDropAddColumn() ?

Reply:
There are different commands: `AlterTableAddCommand` and `CreateTableCommand` and nullability flag is processed by them differently (see `DdlCommandHandler#handle`). I think it's better to check that nullablility is not lost in both cases.